File: Src/java/engine/src/main/java/org/opencds/cqf/cql/engine/exception/InvalidConversion.java
package org.opencds.cqf.cql.engine.exception;

public class InvalidConversion extends CqlException {
    private static final long serialVersionUID = 1L;

    public InvalidConversion(String message) {
        super(message);
    }

    public InvalidConversion(Object from, Object to) {
        super(String.format("Cannot Convert a value of type %s as %s.", from.getClass().getName(), to.getClass().getName()));
    }
}

File: Src/java/engine/src/main/kotlin/org/opencds/cqf/cql/engine/exception/InvalidConversion.kt
package org.opencds.cqf.cql.engine.exception

class InvalidConversion : CqlException {
    constructor(message: String?) : super(message)

    constructor(
        from: Any,
        to: Any,
    ) : super(
        "Cannot Convert a value of type ${from.javaClass.getName()} as ${to.javaClass.getName()}."
    )

    companion object {
        private const val serialVersionUID = 1L
    }
}
